pvm_2_videos.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime
+import os
+from pythonosc import dispatcher, osc_server
+import argparse
+from omxplayer.player import OMXPlayer
+import logging
+from logging.handlers import TimedRotatingFileHandler
+import sys
+
+def _init_logger():
+	logger = logging.getLogger("PVM")
+	logger.setLevel(logging.INFO)
+	handler = logging.StreamHandler(sys.stderr)
+	fileHandler = TimedRotatingFileHandler('./log/{:%Y-%m-%d %H:%M:%S}.log'.format(datetime.now()),  when='midnight')
+	handler.setLevel(logging.INFO)
+	formatter = logging.Formatter("%(asctime)s;%(levelname)s;%(message)s",
+                              "%Y-%m-%d %H:%M:%S")
+	fileHandler.setFormatter(formatter)
+	fileHandler.suffix = '%Y_%m_%d.log'
+	logger.addHandler(fileHandler)
+	handler.setFormatter(formatter)
+	logger.addHandler(handler)
+
+_init_logger()
+_logger = logging.getLogger("PVM")
+_logger.info("Logging system initilized in %s", os.getcwd())
+
+# Place your videos in this folder for autostart
+PEFIX_PATH = "/home/pi/Videos/"
+VIDEO_PATH_ONE = "jellyfish720p.mp4"
+VIDEO_PATH_TWO = "jellyfish720p.mp4"
+media1 = ""
+media2 = ""
+IS_FILE_SET = False
+
+def parse_commands(*args):
+	global media1
+	global media2
+	global VIDEO_PATH_ONE
+	global VIDEO_PATH_TWO
+	global IS_FILE_SET
+	command = args[1]
+	_logger.info("Received command: %s", command)
+	if len(args)>2:
+		value = args[2]
+		_logger.info("Received command: %s", str(value))
+		pass
+	# TODO: Create another python file to control two display
+	if command=="file":
+		if "|" not in value:
+			_logger.info("Please pass two video name like 1.mp4,2.mp4.")
+			return
+		VIDEO_PATH_ONE, VIDEO_PATH_TWO = value.split("|")
+		_logger.info("File set: %s, %s", PEFIX_PATH + VIDEO_PATH_ONE, PEFIX_PATH + VIDEO_PATH_TWO)
+		IS_FILE_SET = True
+		media1 = OMXPlayer(PEFIX_PATH + VIDEO_PATH_ONE, dbus_name='org.mpris.MediaPlayer2.omxplayer1', args=['--loop', '--display', '2'])
+		media1.pause()
+		media2 = OMXPlayer(PEFIX_PATH + VIDEO_PATH_TWO, dbus_name='org.mpris.MediaPlayer2.omxplayer2', args=['--loop', '--display', '7'])
+		media2.pause()
+		return
+
+	if not IS_FILE_SET:
+		_logger.info("Command %s failed because of the file is unset.", command)
+		return
+
+	if command=="start":
+		if media1.can_play():
+			# check two videos' position
+			pos1=media1.position()
+			pos2=media2.position()
+			offset =abs(pos1-pos2)
+			if(offset>1): #more than 1 second offset, direct jump to correct position
+				media2.set_position(pos1)
+			
+			media1.play()
+			media2.play()
+			_logger.info("%s command success.", command)
+		else:
+			_logger.info("%s command failed.", command)
+	elif command=="stop":
+		if media1.can_quit():
+			media1.quit()
+			media2.quit()
+			IS_FILE_SET = False
+			_logger.info("%s command success and file has been unset.", command)
+		else:
+			_logger.info("%s command failed.", command)
+	elif command=="set_position":
+		media1.set_position(float(value))
+		media2.set_position(float(value))
```

Comment:
here's what would be cool to happen here, assuming that the user has already set the device to work with 2 videofiles/players:
- if user sends `set_position` with only 1 argument then set position to both, if he sends it with 2 arguments then set each one to their respective position.
- in alternative (or a next iteration), because the user might want to change the position of just the first video or just the second without affecting the other, it could be required to always send 2 values to affect the 2 players. PLus:
  - if only 1 argument, only the first player is affected,
  - if one wants to affect the second player and just leave the first players with whatever it is without affecting it, then we might need to come up with a special symbol for signifying that bypass. For example if user sends `set_position pass 2` it means that it will not do anything to player one and change player two to position 2.

Makes sense?

Reply:
Yes, it makes sense. My approach for this use case will be the same as the two videos' filenames. For example, 2|3.

Use case:
User input: rate 2|3
Log output: 1.mp4 rate 2, 2.mp4 rate 3

Does this example is you want?

Reply:
Defined in #49